app/helpers/searchHelpers.php

```diff
@@ -432,3 +432,166 @@ function caGetChangeLogForElasticSearch($po_db, $pn_table_num, $pn_row_id) {
+				$vb_b_is_main_table = $vs_b_table === null || $vs_b_table === $vs_table;
+				if ($vb_a_is_main_table && $vb_b_is_main_table) {
+					// Both (a, b) are in the main table, so sort alphabetically by label.
+					return strcasecmp($po_a['label'], $po_b['label']);
+				} elseif (!$vb_a_is_main_table && !$vb_b_is_main_table) {
+					// Both (a, b) are in other tables, so sort alphabetically by table.
+					return strcasecmp($vs_a_table, $vs_b_table);
+				} else {
+					// One of (a, b) is in the main table and the other isn't, so put the one in the main table first.
+					return $vb_a_is_main_table ? -1 : 1;
+				}
+			}
+		});
+		return $va_filters;
+	}
+	# ---------------------------------------
+	function caMapBundleToQueryBuilderFilterDefinition(BaseModel $t_subject, $pa_bundle, Configuration $vo_query_builder_config) {
+		$vs_name = $pa_bundle['bundle'];
+		$vs_name_no_table = preg_replace('/^.*\./', '', $vs_name);
+		$vs_table = $t_subject->tableName();
+		$va_priority = $vo_query_builder_config->get('query_builder_priority_' . $vs_table);
+		$vo_operators_by_type = $vo_query_builder_config->get('query_builder_operators');
+		$vo_field_info = $t_subject->getFieldInfo(substr($vs_name, strpos($vs_name, '.') + 1));
+		$va_element_codes = (method_exists($t_subject, 'getApplicableElementCodes') ? $t_subject->getApplicableElementCodes(null, false, false) : array());
+		$vn_display_type = null;
+		$vs_list_code = null;
+		$va_select_options = null;
+		$va_result = array(
+			'id' => $vs_name,
+			'label' => $pa_bundle['label']
+		);
+		if ($vo_field_info) {
+			// Get the list code and display type for further processing below.
+			$vs_list_code = $vo_field_info['LIST'] ?: $vo_field_info['LIST_CODE'];
+			$vn_display_type = $vo_field_info['DISPLAY_TYPE'];
+			// The "hardcoded" options are `label` => `id` so this needs to be flipped for the query builder.
+			$va_select_options = is_array($vo_field_info['OPTIONS']) ? array_flip($vo_field_info['OPTIONS']) : null;
+			// Convert CA field type to query builder type and operators.
+			switch ($vo_field_info['FIELD_TYPE']) {
+				case FT_NUMBER:
+					$va_result['type'] = 'integer';
+					break;
+				case FT_DATE:
+				case FT_DATERANGE:
+				case FT_HISTORIC_DATE:
+				case FT_HISTORIC_DATERANGE:
+					$va_result['type'] = 'date';
+					break;
+				case FT_TIME:
+				case FT_TIMECODE:
+				case FT_TIMESTAMP:
+				case FT_TIMERANGE:
+					$va_result['type'] = 'time';
+					break;
+				case FT_DATETIME:
+				case FT_HISTORIC_DATETIME:
+					$va_result['type'] = 'datetime';
+					break;
+				default:
+					$va_result['type'] = 'string';
+			}
+		} elseif (in_array($vs_name_no_table, $va_element_codes)) {
+			$t_element = ca_metadata_elements::getInstance($vs_name_no_table);
+			if ($t_element) {
+				// Get the list code and display type for further processing below.
+				$vs_list_code = $t_element->get('list_id');
+				$vn_display_type = $vs_list_code ? DT_SELECT : DT_FIELD;
```

Comment:
For some reason preferred_labels (Object Titles) are coming through as a `<select/>`.